packages/snap/src/rpc/vc/createVP.ts

```diff
@@ -1,24 +1,52 @@
 import type { CreateVPRequestParams } from '@blockchain-lab-um/masca-types';
 import { BIP44CoinTypeNode } from '@metamask/key-tree';
-import type { VerifiablePresentation } from '@veramo/core';
+import type {
+  UnsignedPresentation,
+  VerifiablePresentation,
+  W3CVerifiableCredential,
+} from '@veramo/core';
 
 import type { ApiParams } from '../../interfaces';
+import { getCurrentDidIdentifier } from '../../utils/didUtils';
 import { veramoCreateVP } from '../../utils/veramoUtils';
 
+async function createUnsignedVP(params: {
+  vcs: W3CVerifiableCredential[];
+  did: string;
+}): Promise<UnsignedPresentation> {
+  const { vcs, did } = params;
+  const unsignedVp: UnsignedPresentation = {
+    holder: did,
+    verifiableCredential: vcs,
```

Comment:
VCs need to be checked similarly to veramo: https://github.com/uport-project/veramo/blob/2ce705680173174e7399c4d0607b67b7303c6c97/packages/credential-eip712/src/agent/CredentialEIP712.ts#L215